### ISSUE TITLE
Make the "you can't move here" overlay look nicer

### ISSUE THREAD

**Issue by stevecotton** [wesnoth/wesnoth#9027]
### Describe the desired feature

When moving a unit, the overlay with horizontal stripes that darkens the unreachable hexes is very functional but not pretty on grass terrain, in other words players first encounter it on terrain where it looks bad. In the terrain that players first see it, a uniform grey would look nicer.

I experimented with making the hex mostly uniform grey:

![darken](https://github.com/wesnoth/wesnoth/assets/101462/d4dffbc7-efe1-4c36-96ac-e0a77b52bbb9)
![image](https://github.com/wesnoth/wesnoth/assets/101462/a630517b-d286-418b-b9b7-7ccb880c3fff)

However, it also needs to be functional in caves, which is the reason that the existing image is stripey. Making the center of the overlay uniform grey looks better on grassland, but fails in caves (ignore the whirlpool, that's #9014).

![image](https://github.com/wesnoth/wesnoth/assets/101462/5d92495d-f477-498d-91ae-95a28bdaab27)

I think a good solution might need to choose an image fitting the terrain on the map, which would get more complex when shroud is involved.

The question of hexes that can be moved to, but are currently in shroud/fog is asked in #2152.

**Comment by Dalas121** [comment]
Very exciting! I agree that the existing overlay isn't especially pretty.

Looking at the standard set by other games, my opinion is that the best solution would be to highlight movable hexes, rather than grey out un-movable hexes. For example:

![ingame-ui-hexes](https://github.com/wesnoth/wesnoth/assets/33790750/48a50b6e-1a18-41da-9b31-419403e167dc)

Some related minor improvements that I feel would also be helpful, though possibly out of the scope of this issue:
- disable the map's "grid overlay" by default. I'm not sure if this is still enabled on new installs or not, but at one point it was, and it's not very attractive.
- highlight villages with team color indicators (grey for unclaimed), with an option to disable. Villages don't visually stand out to new players, making them easy to miss (particularly for "unusual" village types like water or windmill).

**Comment by stevecotton** [comment]
Do you have an implementation in game for that image, or is it completely mocked up in an image editor?

**Comment by Dalas121** [comment]
Just mocked up in an editor, I'm afraid

**Comment by gfgtdf** [comment]
I agree that the current overlay doesn't look very good. When i was new to the game i actually thought the stripes were a visual glitch rather than the intended look.

**Comment by beetlenaut** [comment]
Those tiles used to be in a uniform gray. The developers changed it to stripes because they needed to. I don't remember what problem the stripes solved, but I remember reading about it in the changelog or forum somewhere. We should look into that before we change it back.

**Comment by stevecotton** [comment]
Without stripes they aren't visible in caves, as shown in the second picture. I'm currently AFK, but that's the reason that I found in the logs.

**Comment by cooljeanius** [comment]
> I feel like highlighting suitable hexes rather than unsuitable ones like Dalas suggested would also help with a long-standing annoyance I've had when looking at particularly bright maps such as snowy ones. The current approach makes it so any time you select a unit, a large share of map suddenly becomes dim, and then when you deselect the unit, it goes back to bright again. I only have mild photosensitivity but I fear this might be a bigger (potentially hazardous) issue for people with higher photosensitivity issues.

Yeah, this kinda ties in with #6133 a bit...

**Comment by eyalroz** [comment]
> I feel like highlighting suitable hexes rather than unsuitable ones like Dalas suggested would also help with a long-standing annoyance I've had when looking at particularly bright maps such as snowy ones. The current approach makes it so any time you select a unit, a large share of map suddenly becomes dim, and then when you deselect the unit, it goes back to bright again. I only have mild photosensitivity but I fear this might be a bigger (potentially hazardous) issue for people with higher photosensitivity issues.

You're swapping one problem for another. The tinting of already-busy areas of a map, full of units and terrain features, makes it overly "busy", which is unaesthetic IMNSHO, and for some people with different kinds of heightened neurological sensitivity - tiring.

So, at the very least this should be something we can toggle. I would even say the existing behavior is better by default, for most users.

In #9680 I also suggested a third alternative, which is only indicating the edge of the movement range, and in a more subtle way. I'm not sure it's a panacea for anything but another possibility to consider.

**Comment by cooljeanius** [comment]
#9292 made some changes in this area
